### PR TITLE
feat: add soil metadata table to support soil selection feature

### PR DESCRIPTION
**Pull request by tm-ruxandra**
## Description

Add the ability to store metadata about soil data for sites in a new "site metadata" model. Expose a mutation for metadata, and include it in the queryable edges for sites.

### Checklist
- [x] Corresponding issue has been opened
- [x] New tests added

### Related Issues
[#2506](https://github.com/techmatters/terraso-mobile-client/issues/2506)

### Verification steps

Run new unit tests. Verify other functionality through UI ticket.

### CODE REVIEW

File: terraso_backend/tests/graphql/test_sites_soil_data_query.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Copyright © 2023 Technology Matters
```
Comment (paulschreiber) [inline terraso_backend/tests/graphql/test_sites_soil_data_query.py:1]:
```suggestion
# Copyright © 2024 Technology Matters
```

File: terraso_backend/tests/graphql/test_sites_soil_data_query.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      }
+    }
+    """
+    client.force_login(site_creator)
+
+    response = graphql_query(query % site.id, client=client)
+
+    assert "errors" not in response.json()
+    site_json = response.json()["data"]["site"]
+    assert site_json["name"] == site.name
```
Comment (paulschreiber) [inline terraso_backend/tests/graphql/test_sites_soil_data_query.py:43]: question: should the test use a literal string here?

File: terraso_backend/tests/graphql/mutations/test_soil_metadata.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
+# Copyright © 2023 Technology Matters
```
Comment (paulschreiber) [inline terraso_backend/tests/graphql/mutations/test_soil_metadata.py:1]:
```suggestion
# Copyright © 2024 Technology Matters
```

File: terraso_backend/apps/soil_id/migrations/0021_soilmetadata.py [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# Generated by Django 5.1.3 on 2024-11-18 20:52
```
Comment (paulschreiber) [inline terraso_backend/apps/soil_id/migrations/0021_soilmetadata.py:1]: issue: add copyright notice.

**Review by knipec — APPROVED** [review]

**Pull request merged**